common/archiver/archivalMetadata.go

```diff
@@ -51,13 +63,54 @@ const (
 	ArchivalEnabled
 )
 
+// NewArchivalMetadata constructs a new ArchivalMetadata
+func NewArchivalMetadata(
+	historyStatus string,
+	historyReadEnabled bool,
+	visibilityStatus string,
+	visibilityReadEnabled bool,
+	domainDefaults *config.ArchivalDomainDefaults,
+) ArchivalMetadata {
+	return &archivalMetadata{
+		historyConfig: NewArchivalConfig(
+			historyStatus,
+			historyReadEnabled,
+			domainDefaults.History.Status,
+			domainDefaults.History.URI,
+		),
+		visibilityConfig: NewArchivalConfig(
+			visibilityStatus,
+			visibilityReadEnabled,
+			domainDefaults.Visibility.Status,
+			domainDefaults.Visibility.URI,
+		),
+	}
+}
+
+func (metadata *archivalMetadata) GetHistoryConfig() *ArchivalConfig {
```

Comment:
Hummm.... its not actually true that ArchivalConfig is immutable the way we are returning it and exporting fields. What was the reason you wanted to remove the getters and return a reference to struct directly?

Reply:
As discussed, created an interface and provided getter methods.